### PR TITLE
feat(substrate): enable deployment with proxy and vault via helm

### LINKED ISSUE #2484: epic: platform deployments using just helm commands and charts
**As Developers, we want to implement deployment of platforms using just helm charts**

**Few of the key improvements in the platform's charts folder as follows:**
- Update `parameters table` to accurately reflect variable hierarchy and use `camelCase` for parameter/variable names.
- Align the READMEs of a specific platform with the structure and design of `BESU` READMEs.
- Implement hook concept to delete all the supported resources of a chart while doing the network reset
- Utilize `helpers.tpl` to assign the deployment name instead of metadata section in template (.tpl) and values.yaml files.
- Organize common configuration variables, like those associated with Vault and Cluster, under the `global` section in template (.tpl) and values.yaml files. Ensure these variables are used with the updated hierarchy within the respective chart.
- Implement chart dependencies and  use pre install hook concept to create prerequisite Kubernetes  resources and run one time tasks  

**Required testing**:
- Perform Helm lint manually to quickly validate the chart, followed by the successful network deployment and reset using just helm charts and commands.
- For any concerns and their solutions, please follow the Besu Platform.

**This issue needs to be fixed for the following platforms**:
- [x] Besu @sownak 
- [x] Quorum @saurabhkumarkardam 
- [x] Fabric @mgCepeda 
- [x] R3-Corda-Ent @suvajit-sarkar 
- [x] R3-Corda @sownak 
- [x] Substrate @saurabhkumarkardam and @saikumarbommakanti

### PR DESCRIPTION
**Pull request by saikumarbommakanti**
This pull request introduces enhanced functionality to deploy the Substrate platform using Helm charts. Several improvements have been made to streamline the process and increase readability:

Following charts supports vault and proxy

1. Substrate-genesis:
2. Substrate-node:
3. dscp-ipfs-node

4. A README.md has been added at the path ./platform/substrate/charts/ to provide guidance for deploying the Substrate platform via Helm.

These enhancements aim to streamline Substrate deployment via Helm, making the process more efficient and the codebase easier to manage.

fixes #2484

### CODE REVIEW

File: platforms/substrate/charts/values/noproxy-and-novault/ipfs.yaml [hunk 1]
```diff
@@ -10,10 +10,10 @@ global:
     provider: azure
     cloudNativeServices: false
   vault:
-    type: kubernetes # kubernetes | hashicorp
+    type: hashicorp # kubernetes | hashicorp
```
Comment (suvajit-sarkar) [inline platforms/substrate/charts/values/noproxy-and-novault/ipfs.yaml:13]: why does no proxy and no vault value file have vault and proxy set ?

File: platforms/substrate/charts/substrate-genesis/templates/generate-keys.yaml [hunk 2]
```diff
@@ -83,6 +87,60 @@ spec:
...
+                  local key="$1"
+                  local payload_json="$2"
+                  
+                  # Read secret from vault
+                  vaultBevelFunc "readJson" "${VAULT_SECRET_ENGINE}/${VAULT_SECRET_PREFIX}/${key}"
+                  # Check if secrets are available in the vault
+                  if [ "$SECRETS_AVAILABLE" == "yes" ]
+                    then
+                      # Extract secrets from JSON response
+                      local node_id=$(echo ${VAULT_SECRET} | jq -r '.["node_id"]')
```
Comment (suvajit-sarkar) [inline platforms/substrate/charts/substrate-genesis/templates/generate-keys.yaml:108]: what does local do here ?

File: platforms/substrate/charts/substrate-genesis/templates/genesis.yaml [hunk 3]
```diff
@@ -195,6 +197,29 @@ spec:
...
+          {{- if eq $.Values.node.image "docker.io/paritytech/substrate-playground-template-node-template" }}
+          export PATH=$PATH:~/workspace
+          {{- else }}
+          mv vault /bin
+          {{- end }}
+          vault --version
+          # Save the generated keys to VAULT
+          if [ "$SECRETS_AVAILABLE" == "no" ]; then
+            vault_secret_key="${VAULT_SECRET_ENGINE}/${CERTS_SECRET_PREFIX}/genesis"
+            vault kv put "$vault_secret_key" genesis=@"${MOUNT_PATH}/genesis_base64" sudo_details=@"${MOUNT_PATH}/sudoKeygenOutput.json"
```
Comment (suvajit-sarkar) [inline platforms/substrate/charts/substrate-genesis/templates/genesis.yaml:220]: why are we downloading and using vault cli ? why not use the bevel vault script ?

**Review by suvajit-sarkar — CHANGES_REQUESTED** [review]

File: platforms/substrate/charts/README.md [hunk 4]
```diff
@@ -64,8 +64,8 @@ node:
   ...
```
Comment (sownak) [inline platforms/substrate/charts/README.md:64]: Instructions to update a value file here should not be given, the value files should already have the right values

File: platforms/substrate/charts/README.md [hunk 5]
```diff
@@ -87,23 +87,79 @@ helm install    member-1 ./substrate-node --namespace supplychain-subs --values
 ```yaml
```
Comment (sownak) [inline platforms/substrate/charts/README.md:87]: same comment as above

File: platforms/substrate/charts/README.md [hunk 6]
```diff
@@ -87,23 +87,79 @@ helm install    member-1 ./substrate-node --namespace supplychain-subs --values
...
+```
+### 2. Install Bootnode
+```bash
+helm install validator-1 ./substrate-node --namespace supplychain-subs --values ./values/proxy-and-vault/validator.yaml --set global.proxy.p2p=15051
+```
+### 3. Install Additional Nodes
+
+To deploy additional nodes, update the following section in the `./values/proxy-and-vault/node.yaml` file only once:
+```yaml
+...
```
Comment (sownak) [inline platforms/substrate/charts/README.md:128]: Same comment. bootnode address update should be mentioned at the start, not in each step

File: platforms/substrate/charts/README.md [hunk 7]
```diff
@@ -87,23 +87,79 @@ helm install    member-1 ./substrate-node --namespace supplychain-subs --values
...
+    bootnodePort: 15051
+  ...
+  ...
+...
+```
+
+# Then install the nodes using the following commands:
+```bash
+
+helm install validator-2 ./substrate-node --namespace supplychain-subs --values ./values/proxy-and-vault/validator.yaml --set global.proxy.p2p=15052
```
Comment (sownak) [inline platforms/substrate/charts/README.md:144]: does the validator use TCPMapping? if not then same ambassador port can be used

File: platforms/substrate/charts/dscp-ipfs-node/templates/statefulset.yaml [hunk 8]
```diff
@@ -167,7 +169,7 @@ spec:
                 jq -r 'if .errors then . else .auth.client_token end')
 
               validateVaultResponse 'vault login token' "${VAULT_CLIENT_TOKEN}"
-              vault_secret_key="${CERTS_SECRET_PREFIX}/ipfs"
+              vault_secret_key="${VAULT_SECRET_ENGINE}"/"${VAULT_SECRET_PREFIX}/ipfs"
```
Comment (sownak) [inline platforms/substrate/charts/dscp-ipfs-node/templates/statefulset.yaml:172]: Why is this not using bevel-scripts?

File: platforms/substrate/charts/substrate-node/values.yaml [hunk 9]
```diff
@@ -170,7 +170,7 @@ node:
   perNodeServices:
     createApiService: true
     createP2pService: true
-    p2pServiceType: ClusterIP # Must be type ClusterIP, NodePort or LoadBalancer, If using type NodePort or LoadBalancer then you must set NodeSelecter accordingly.
+    p2pServiceType: LoadBalancer # Must be type ClusterIP, NodePort or LoadBalancer, If using type NodePort or LoadBalancer then you must set NodeSelecter accordingly.
```
Comment (sownak) [inline platforms/substrate/charts/substrate-node/values.yaml:173]: why is this loadbalancer?

**Review by sownak — CHANGES_REQUESTED** [review]

File: platforms/substrate/charts/README.md (hunk 7 above)
Comment (saikumarbommakanti) [reply platforms/substrate/charts/README.md:144]: yes it uses tcpmapping

File: platforms/substrate/charts/dscp-ipfs-node/templates/statefulset.yaml (hunk 8 above)
Comment (saikumarbommakanti) [reply platforms/substrate/charts/dscp-ipfs-node/templates/statefulset.yaml:172]: Already checked the script it is not working, That is why we are having code here it self

**Pull request merged**